### ISSUE TITLE
[Feature]: update OpenCL-kernel

### ISSUE THREAD

**Issue by kitsudaiki** [kitsudaiki/ainari#108]
### Description

The last recent changes were only tested with CPU and CUDA-kernel. The OpenCL-kernel compatible with the current data-structures. This has to be fixed by porting the CUDA-kernel to OpenCL.

### Acceptance Criteria

1. The OpenCL-kernel works the same way with the same accuracy like the CUDA-kernel
2. There should be an automatic detection, if the available gpu is an nvidia-gpu or not
3. The OpenCL-kernel is used instead of the CUDA-kernel, when no nvidia-gpu is available
4. For the purpose of testing, it is possible to override the automatic detection and enforce the OpenCL-kernel even on nvidia-gpus

### Additional Information

In Version 0.2.0 there was already a working version. Check this one out 

### Blocked by

- #61 because `cmake` privides a better option to compile text-files into the code

### How to test

- Test on node with OpenCL- and CUDA-kernel and if possible test on a node with AMD and a node with NVIDIA gpu

**Comment by kitsudaiki** [comment]
At the moment this is only a nice-to-have feature and not necessary for now, so it was moved into the backlog

**Comment by kitsudaiki** [comment]
deprecated because of https://github.com/kitsudaiki/ainari/issues/906
